### ISSUE TITLE
spoof location

### ISSUE THREAD

**Issue by vixalien** [revanced/revanced-patches#526]
### Application

Youtube Music

### Issue

it should fix YouTube music not working in certain countries

### Patch

Allow selection of a country to be spoofed as so that youtube music works

### Motivation

because some of us can't access YouTube music because it's not available. with this patch, it would make YouTube music available for more people

**Comment by webzeroacs** [comment]
Are you talking about this?
![Cap 2022-09-18 23-48-45-109](https://user-images.githubusercontent.com/20047139/190913161-27b8a077-0d71-4575-9df7-c59b877926da.png)

I'd love to have this function.

**Comment by vixalien** [comment]
yeah. that's exactly what i'm talking about

**Comment by vixalien** [comment]
yes i believe that's how it's done. what do you mean it's not possible?? can't patches set up a mini proxy that only intercept and spoof the location and nothing else??

**Comment by vixalien** [comment]
yes i believe that's how it's done. what do you mean it's not possible?? can't patches set up a mini proxy that only intercept and spoof the location and nothing else??

**Comment by oSumAtrIX** [comment]
Still requires a proxy, at that point it's easier to use a vpn.

**Comment by vixalien** [comment]
a vpn will encapsulate all my apps. the patch would only be for yt.

**Comment by EatVanced** [comment]
who's going to pay the proxy cost? You could use services like ControlD or probably Smart DNS Proxy, that you can set to traffic youtube only, even then, some vpn app allow you to make it only certain apps would use the vpn connection.

**Comment by oSumAtrIX** [comment]
If this feature gets implemented, you will have to specify your own proxy.

**Comment by vixalien** [comment]
Isn't there a way to have code that just interprets the network calls 
in vanced music and only affect the needed one. eg:

```js
const oldFetch = globablThis.fetch;

function fetch  (url, options) {
	if (url === "<https://some.geolocati.on/website>") {
		return json({ lat: -1.9, long: 40.1 });
	} else {
		// passthrough
		return 	oldFetch(url, options);
	}
}

globalThis.fetch = fetch;
```

this is what i was calling a proxy. Isn't there a way to implement this 
(that's the general idea of one would implement that in JavaScript).

**Comment by oSumAtrIX** [comment]
YouTube Music has an entire object oriented and highly abstracted implementation of their networking infrastructure, it's not as trivial as a simple web request.

**Comment by vixalien** [comment]
I get it. I thought it was plausible. Thanks btw.

 > Or isn't there a way to patch the library that it uses under the 
hood?

**Comment by Fooftilly** [comment]
I used Orbot to circumvent IP requirements. I would be nice if you could use Tor as a proxy directly in the app.

**Comment by ilyassesalama** [comment]
This definitely should be implemented if possible.

**Comment by pixiekat** [comment]
> a vpn will encapsulate all my apps. the patch would only be for yt.

There are several VPNs on Android that allow split tunnelling

**Comment by vixalien** [comment]
vpns are paid and you can never be sure of the security of the free ones.

also, vpns make the network slower and this would be a cleaner solution imo

**Comment by lordcyb3r** [comment]
> vpns are paid and you can never be sure of the security of the free ones.
> 
> also, vpns make the network slower and this would be a cleaner solution imo

Most likely a non issue when it comes to music. The music stream is compressed and doesnt have a huge bandwidth footprint. I personally suggest mullvad vpn. 5 euros a month, solid privacy policy, and split tunneling.

**Comment by ILoveOpenSourceApplications** [comment]
So is this issue solvable or not?

**Comment by oSumAtrIX** [comment]
It is, but it likely wont be unless anyone is willing to proxy web requests

**Comment by UnlimitedCookies** [comment]
Sounds like a `proxy all app traffic through a proxy` (with a proxy specified in settings) patch, which sounds hard to implement (might be more a thing that a custom ROM would do).

**Comment by ILoveOpenSourceApplications** [comment]
RVX just implemented spoof-version patch which fixes this issue. Can someone look into that and see whether we can implement it as well?
